### PR TITLE
[GLU-235] Composable Banners + Presets

### PR DESCRIPTION
**Pull request by mboperator**
## Changes
- Breaks Icon out of Banner
- Introduces ErrorBanner, SuccessBanner, and InfoBanner

### CODE REVIEW

File: packages/core-react/src/components/Banner/presets.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Banner from './Banner'
+import Icon from '../Icon'
```
Comment (farism) [inline packages/core-react/src/components/Banner/presets.js]: Do you need this import if you are using `Banner.Icon`?

File: packages/core-react/src/components/Banner/presets.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const InfoBanner = ({ children, ...props }) => (
+  <Banner variant="info">
+    <Banner.Icon variant="info" />
+    {children}
+  </Banner>
+)
+
+InfoBanner.propTypes = {
+  children: PropTypes.node,
+}
```
Comment (farism) [inline packages/core-react/src/components/Banner/presets.js]: Do we need an `ActionBanner` variant?

File: packages/core-react/src/components/Banner/presets.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Banner from './Banner'
+import Icon from '../Icon'
+
+export const ErrorBanner = ({ children, ...props }) => (
```
Comment (farism) [inline packages/core-react/src/components/Banner/presets.js]: Should we be splatting down `{...props}`? That would be forward compatible if we ever add more props to `Banner`

File: packages/core-react/src/components/Banner/story.js [hunk 4]
```diff
@@ -3,12 +3,13 @@ import { storiesOf } from '@storybook/react'
 import { action } from '@storybook/addon-actions'
 
 import Banner from './'
+import { InfoBanner, ErrorBanner, SuccessBanner } from './presets'
```
Comment (farism) [inline packages/core-react/src/components/Banner/story.js]: n/b maybe name this file `./variants`?

File: packages/core-react/src/components/Banner/presets.js (hunk 2 above)
Comment (mboperator) [reply packages/core-react/src/components/Banner/presets.js]: Added

**Pull request merged**